kura/test/org.eclipse.kura.nm.test/src/test/java/org/eclipse/kura/nm/configuration/NetworkPropertiesTest.java

```diff
@@ -0,0 +1,288 @@
+		thenResultEquals(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3"));
+	}
+	
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedStringNull() {
+		givenTheMapWith("testKey-comma-seperated", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated");
+		thenAnNoSuchElementExceptionOccured();
+	}
+	
+	@Test
+	public void shouldReturnStringListGetStringListWithCommaSeperatedStringMissing() {
+		givenTheMapWith("testKey-comma-seperated", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey-comma-seperated-not-existant");
+		thenAnNoSuchElementExceptionOccured();
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithRegularString() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKey1");
+		thenResultEquals(Arrays.asList("testString1"));
+	}
+
+	@Test
+	public void shouldReturnStringListGetStringListWithNull() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListIsCalledWith("testKeyNull");
+		thenResultEquals(Arrays.asList());
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithCommaSeperatedString() {
+		givenTheMapWith("testKey-comma-seperated", "commaSeperated1,commaSeperated2,commaSeperated3");
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(Optional.of(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3")));
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithCommaSeperatedStringMalformed() {
+		givenTheMapWith("testKey-comma-seperated",
+				", , ,,,,commaSeperated1, , , ,,,,,commaSeperated2,,,, ,, ,,commaSeperated3,, , ,,,, ,");
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKey-comma-seperated");
+		thenResultEquals(Optional.of(Arrays.asList("commaSeperated1", "commaSeperated2", "commaSeperated3")));
+	}
+
+	@Test
+	public void shouldReturnOptionalStringListGetStringListOptWithString() {
+		givenTheMapWith("testKey1", "testString1");
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKey1");
+		thenResultEquals(Optional.of(Arrays.asList("testString1")));
+	}
+
+	@Test
+	public void shouldReturnEmptyGetStringListOptWithNull() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("testKeyNull");
+		thenResultEquals(Optional.empty());
+	}
+
+	@Test
+	public void shouldReturnEmptyGetStringListOptWithNoKey() {
+		givenTheMapWith("testKeyNull", null);
+		givenNetworkPropsIsCreated();
+		whenGetStringListOptIsCalledWith("");
+		thenResultEquals(Optional.empty());
+	}
+
+	/*
+	 * Given
+	 */
+
+	public void givenNetworkPropertiesBuiltWith(Map<String, Object> properties) {
+		this.properties = properties;
+	}
+
+	public void givenTheMapWith(String key, Object pair) {
```

Comment:
Finally... this is just a style preference, what do you think of:

```suggestion
	public void givenMapWith(String key, Object pair) {
```

Reply:
I keep saying "finally" but we both know I'm lying XD